ffi/src/scan.rs

```diff
@@ -112,27 +136,34 @@ unsafe fn scan_impl(
             scan_builder = scan_builder.with_predicate(predicate);
         }
     }
-    Ok(BoxHandle::into_handle(scan_builder.build()))
+    Ok(Box::new(scan_builder.build()).into())
 }
 
-impl BoxHandle for GlobalScanState {}
+pub struct GlobalScanStateHandle {
```

Comment:
```suggestion
pub struct MutGlobalScanState {
```

Not sure what prefix we want to use instead of "Shared" for `Mutable = True` things. Maybe `Mut`, or maybe `Exclusive`?